aspects/aspects.go

```diff
@@ -530,6 +547,75 @@ func schemaTypesStr(types []SchemaType) string {
 	return sb.String()
 }
 
+// getValuesThroughPaths takes a match's storage path and unmatched suffix and
+// strips the outer layers of the value to be set so it can be used at the storage
+// path. Parts of the suffix that are placeholders will be expanded based on what
+// keys exist in the value at that point and the mapping will be used to complete
+// the storage path.
+var getValuesThroughPaths = getValuesThroughPathsImpl
+
+func getValuesThroughPathsImpl(storagePath string, suffixParts []string, val interface{}) (map[string]interface{}, error) {
```

Comment:
are suffixParts form a storage path or  a request path? the requestPathi I think maybe it shoud have req in the name?

Reply:
The suffixParts is the unmatched suffix of the request path that we then use to strip out outer layers of the value. I'll change to `reqSuffixParts` then